app/services/stories_api/v3/endpoints/story.rb

```diff
@@ -35,7 +36,12 @@ def get
         end
 
         def patch
-          story = user.user_sets.custom_find(params[:id])
+          story = if RecordSchema.roles[@user.role.to_sym].try(:admin)
```

Comment:
The `admin?` method does the same thing.

https://github.com/DigitalNZ/supplejack_api/blob/master/app/models/supplejack_api/user.rb#L199

Or can you use `can_change_featured_sets?` method. https://github.com/DigitalNZ/supplejack_api/blob/master/app/models/supplejack_api/user.rb#L203